### PR TITLE
:running: Add a validation check in place against client-go@incompatible

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
This sneaks in way too often, let's add a validation check for it

/milestone v0.3.0

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -537,6 +537,10 @@ verify-modules: modules
 		git diff; \
 		echo "go module files are out of date"; exit 1; \
 	fi
+	@if (find . -name 'go.mod' -print0 | xargs -n1 grep -q -i 'k8s.io/client-go.*+incompatible'); then \
```
Comment (ncdc) [inline Makefile]: Do you need to escape the `+`?

Comment (vincepri) [reply Makefile]: if I do, it says `grep: repetition-operator operand invalid`, so I guess no?

Comment (ncdc) [reply Makefile]: Oh this isn't egrep, ok

**Pull request merged**